Logger/LogFactory.cs

```diff
@@ -2,10 +2,22 @@
 {
     public class LogFactory
     {
+        private string? FilePath;
+
         public BaseLogger CreateLogger(string className)
         {
-            
-            return null;
+            if(FilePath is null)
+            {
+                return null!;
+            }
+
+            FileLogger output = new FileLogger(FilePath) {ClassName = className};
+            return output;
```

Comment:
```suggestion
            return new FileLogger(FilePath) {ClassName = className};
```
I'd recommend simplifying your code to the above.